pkg/action/install.go

```diff
@@ -368,63 +371,76 @@ func (i *Install) RunWithContext(ctx context.Context, chrt *chart.Chart, vals ma
 		// not working.
 		return rel, err
 	}
-	rChan := make(chan resultMessage)
-	ctxChan := make(chan resultMessage)
-	doneChan := make(chan struct{})
-	defer close(doneChan)
-	go i.performInstall(rChan, rel, toBeAdopted, resources)
-	go i.handleContext(ctx, ctxChan, doneChan, rel)
-	select {
-	case result := <-rChan:
-		return result.r, result.e
-	case result := <-ctxChan:
-		return result.r, result.e
+
+	rel, err = i.performInstallCtx(ctx, rel, toBeAdopted, resources)
+	if err != nil {
+		rel, err = i.failRelease(rel, err)
 	}
+	return rel, err
 }
 
-func (i *Install) performInstall(c chan<- resultMessage, rel *release.Release, toBeAdopted kube.ResourceList, resources kube.ResourceList) {
+func (i *Install) performInstallCtx(ctx context.Context, rel *release.Release, toBeAdopted kube.ResourceList, resources kube.ResourceList) (*release.Release, error) {
+	type Msg struct {
+		r *release.Release
+		e error
+	}
+	resultChan := make(chan Msg, 2)
+	doneChan := make(chan struct{})
+	defer close(doneChan)
+
+	go func() {
+		rel, err := i.performInstall(rel, toBeAdopted, resources)
+		resultChan <- Msg{rel, err}
+	}()
+	go func() {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			resultChan <- Msg{rel, err}
+		case <-doneChan:
+			return
+		}
+	}()
```

Comment:
Think this whole bit (and below) is an obsolete routine? AFAIK, the following should work without any issues while yielding the same behavior and getting rid of a `chan`:

```go
select {
case <-ctx.Done():
	return rel, ctx.Err()
case msg := <-resultChan:
	return msg.r, msg.e
}